marshal/marshal.go

```diff
@@ -0,0 +1,185 @@
+package marshal
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"golang.org/x/exp/maps"
+)
+
+const forceSendFieldName = "ForceSendFields"
+
+// Marshal returns a JSON encoding of the given object. Included fields:
+// - non-empty value
+// - a basic type whose field's name is present in forceSendFields
+// Embedded structs are still considered a separate struct. ForceSendFields
+// in an embedded struct only impact the fields of the embedded struct.
+// Conversely, an embedded struct is not impacted by the ForceSendFields
+// of the struct containing it.
+func Marshal(object any) ([]byte, error) {
```

Comment:
Add comment about Interface/pointer values that are not serialized, even if ForceSendFields includes their field name.